client/state/happychat/middleware.js

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_TRANSCRIPT_REQUEST,
+} from 'state/action-types';
+
+import { connection } from './common';
+
+import { receiveChatTranscript } from './actions';
+import { getHappychatTranscriptTimestamp } from './selectors';
+
+const debug = require( 'debug' )( 'calypso:happychat:actions' );
+
+export const requestTranscript = ( { getState, dispatch } ) => {
```

Comment:
Would it be possible to add a unit test for this? I'm wondering if we could mock the socket connection like we do with REST APIs with `nock`. Either way, I think it might worth another PR itself, so this is more of an inquiry : )